integrated_channels/blackboard/client.py

```diff
@@ -525,37 +552,49 @@ def _get_or_create_integrated_grade_column(self, bb_course_id, grade_column_name
             bb_course_id (str): The Blackboard course ID in which to search for the edX final grade,
             grade column.
         """
-        grade_column_response = self._get(self.generate_gradebook_url(bb_course_id))
-        parsed_response = grade_column_response.json()
-        grade_columns = parsed_response.get('results')
-
+        gradebook_column_url = self.generate_gradebook_url(bb_course_id)
         grade_column_id = None
-        for grade_column in grade_columns:
-            if grade_column.get('externalId') == external_id:
-                grade_column_id = grade_column.get('id')
-        if not grade_column_id:
-            # Potential customization here per-customer, if the need arises.
-            grade_column_data = {
-                "externalId": external_id,
-                "name": grade_column_name,
-                "displayName": grade_column_name,
-                "description": "edX learner's grade.",
-                "externalGrade": False,
-                "score": {
-                    "possible": points_possible
-                },
-                "availability": {
-                    "available": "Yes"
-                },
-                "grading": {
-                    "type": "Manual",
-                    "scoringModel": "Last",
-                    "anonymousGrading": {
-                        "type": "None",
-                    }
-                },
-            }
+        more_pages_present = True
+        current_page_count = 0
+
+        while more_pages_present and current_page_count <= 250:
+            grade_column_response = self._get(gradebook_column_url)
+            parsed_response = grade_column_response.json()
+            grade_columns = parsed_response.get('results')
+
+            for grade_column in grade_columns:
+                if grade_column.get('externalId') == external_id:
+                    grade_column_id = grade_column.get('id')
+                    break
+
+            # Blackboard's pagination is returned within the response json if it exists
+            # Example:
+            # Response = {
+            #     'results': [{
+            #         'id': 1,
+            #         ...
+            #     }, {
+            #         'id', 2,
+            #         ...
+            #     }],
+            #     'paging': {
+            #         'nextPage': '/learn/api/public/v1/courses/<courseID>/gradebook/columns?offset=200'
+            #     }
+            # }
+            if parsed_response.get('paging') and not grade_column_id:
```

Comment:
can you ELI5 why grade_column_id not existing affects whether we check for more pages or not. I did read the description, but I may be too hazy to comprehend it if it was mentioned there.

Reply:
Is it that we go through pages of results until we get a grade_column_id and then we stop going through the pages?

Reply:
Yeah the idea is that we either go through every page or we find the ID on a particular page. Blackboard will only include the `paging` response if there exists another page, so that line is saying "if we haven't found ID yet and if there are more pages available".